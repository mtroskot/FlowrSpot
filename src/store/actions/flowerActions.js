import { flowerActionTypes } from 'src/constants/actionTypes';

export const getFavoriteFlowers = () => ({
  type: flowerActionTypes.GET_FAVORITE_FLOWERS,
  payload: {}
});

export const getFavoriteFlowersSuccess = favoriteFlowers => ({
  type: flowerActionTypes.GET_FAVORITE_FLOWERS_SUCCESS,
  payload: { favoriteFlowers }
});

export const favoriteFlower = flowerId => ({
  type: flowerActionTypes.FAVORITE_FLOWER,
  payload: { flowerId }
});

export const favoriteFlowerSuccess = (flowerId, favoriteId) => ({
  type: flowerActionTypes.FAVORITE_FLOWER_SUCCESS,
  payload: { flowerId, favoriteId }
});

export const unfavoriteFlower = (flowerId, favoriteId) => ({
  type: flowerActionTypes.UNFAVORITE_FLOWER,
  payload: { flowerId, favoriteId }
});

export const unfavoriteFlowerSuccess = flowerId => ({
  type: flowerActionTypes.UNFAVORITE_FLOWER_SUCCESS,
  payload: { flowerId }
});
