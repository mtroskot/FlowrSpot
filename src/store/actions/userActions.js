import { userActionTypes } from 'src/constants/actionTypes';

export const registerUser = registerData => ({
  type: userActionTypes.REGISTER_USER,
  payload: { registerData }
});

export const loginUser = (email, password) => ({
  type: userActionTypes.LOGIN_USER,
  payload: { email, password }
});

export const loginUserSuccess = (id, email, authToken, firstName, lastName) => ({
  type: userActionTypes.LOGIN_USER_SUCCESS,
  payload: { id, email, authToken, firstName, lastName }
});

export const logoutUser = () => ({
  type: userActionTypes.LOGOUT_USER,
  payload: {}
});
