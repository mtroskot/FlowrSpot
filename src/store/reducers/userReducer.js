import { userActionTypes } from 'src/constants/actionTypes';

const initialState = {
  id: null,
  email: null,
  authToken: null,
  firstName: null,
  lastName: null
};

const userReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case userActionTypes.LOGIN_USER_SUCCESS:
      return {
        ...state,
        id: payload.id,
        email: payload.email,
        authToken: payload.authToken,
        firstName: payload.firstName,
        lastName: payload.lastName
      };
    default:
      return state;
  }
};

export default userReducer;
