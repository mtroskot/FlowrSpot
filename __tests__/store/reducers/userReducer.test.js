import userReducer from 'src/store/reducers/userReducer';
import * as userActions from 'src/store/actions/userActions';

const initialState = {
  id: null,
  email: null,
  authToken: null,
  firstName: null,
  lastName: null
};

describe('INITIAL_STATE', () => {
  const beforeActionState = initialState;

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(userReducer(beforeActionState, action)).toEqual(expectedState);
    expect(userReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns loginUserSuccess state', () => {
    const id = 'id';
    const email = 'email';
    const authToken = 'authToken';
    const firstName = 'firstName';
    const lastName = 'lastName';
    const loginAction = userActions.loginUserSuccess(id, email, authToken, firstName, lastName);
    const expectedState = { ...beforeActionState, id, email, authToken, firstName, lastName };
    expect(userReducer(beforeActionState, loginAction)).toEqual(expectedState);
    expect(userReducer(beforeActionState, loginAction)).toMatchSnapshot();
  });
});

describe('LOGIN_USER_SUCCESS STATE', () => {
  const beforeActionState = {
    ...initialState,
    id: 'id',
    email: 'email',
    authToken: 'authToken',
    firstName: 'firstName',
    lastName: 'lastName'
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(userReducer(beforeActionState, action)).toEqual(expectedState);
    expect(userReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns loginUserSuccess state', () => {
    const id = 'id2';
    const email = 'email2';
    const authToken = 'authToken2';
    const firstName = 'firstName2';
    const lastName = 'lastName2';
    const loginAction = userActions.loginUserSuccess(id, email, authToken, firstName, lastName);
    const expectedState = { ...beforeActionState, id, email, authToken, firstName, lastName };
    expect(userReducer(beforeActionState, loginAction)).toEqual(expectedState);
    expect(userReducer(beforeActionState, loginAction)).toMatchSnapshot();
  });
});
