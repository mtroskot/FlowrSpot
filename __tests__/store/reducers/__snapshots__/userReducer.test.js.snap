// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL_STATE 1returns initial state 1`] = `
Object {
  "authToken": null,
  "email": null,
  "firstName": null,
  "id": null,
  "lastName": null,
}
`;

exports[`INITIAL_STATE 2returns loginUserSuccess state 1`] = `
Object {
  "authToken": "authToken",
  "email": "email",
  "firstName": "firstName",
  "id": "id",
  "lastName": "lastName",
}
`;

exports[`LOGIN_USER_SUCCESS STATE 1returns initial state 1`] = `
Object {
  "authToken": "authToken",
  "email": "email",
  "firstName": "firstName",
  "id": "id",
  "lastName": "lastName",
}
`;

exports[`LOGIN_USER_SUCCESS STATE 2returns loginUserSuccess state 1`] = `
Object {
  "authToken": "authToken2",
  "email": "email2",
  "firstName": "firstName2",
  "id": "id2",
  "lastName": "lastName2",
}
`;
